src/agent/src/watcher.rs

```diff
@@ -0,0 +1,779 @@
+                    if entry.target_mount_point.as_path().is_dir() {
+                        for dir_entry in std::fs::read_dir(entry.target_mount_point.as_path())? {
+                            let dir_entry = dir_entry?;
+                            let path = dir_entry.path();
+                            if dir_entry.file_type()?.is_dir() {
+                                tokio::fs::remove_dir_all(path).await?;
+                            } else {
+                                tokio::fs::remove_file(path).await?;
+                            }
+                        }
+                    }
+
+                    //  - Create bind mount from source to destination
+                    BareMount::new(
+                        entry.source_mount_point.to_str().unwrap(),
+                        entry.target_mount_point.to_str().unwrap(),
+                        "bind",
+                        MsFlags::MS_BIND,
+                        "bind",
+                        logger,
+                    )
+                    .mount()?;
+                }
+                Ok(count) => {
+                    debug!(logger, "watched {} files", count);
+                }
+            };
+        }
+
+        Ok(())
+    }
+}
+
+/// Handles watchable mounts. The watcher will manage one or more mounts for one or more containers. For each
+/// mount that is added, the watcher will maintain a list of files to monitor, and periodically checks for new,
+/// removed or changed (modified date) files. When a change is identified, the watcher will either copy the new
+/// or updated file to a target mount point, or remove the removed file from the target mount point.  All WatchableStorage
+/// target mount points are expected to reside within a single tmpfs, whose root is created by the BindWatcher.
+///
+/// This is a temporary workaround to handle config map updates until we get inotify on 9p/virtio-fs.
+/// More context on this:
+/// - https://github.com/kata-containers/runtime/issues/1505
+/// - https://github.com/kata-containers/kata-containers/issues/1879
+#[derive(Debug, Default)]
+pub struct BindWatcher {
+    /// Container ID -> Vec of watched entries
+    shared: Arc<Mutex<HashMap<String, Entries>>>,
+    watch_thread: Option<task::JoinHandle<()>>,
+}
+
+impl Drop for BindWatcher {
+    fn drop(&mut self) {
+        self.cleanup();
+    }
+}
+
+impl BindWatcher {
+    pub fn new() -> BindWatcher {
+        Default::default()
+    }
+
+    pub async fn add_container(
+        &mut self,
+        id: String,
+        mounts: impl IntoIterator<Item = Storage>,
+        logger: &Logger,
+    ) -> Result<()> {
+        if self.watch_thread.is_none() {
+            // Virtiofs shared path is RO by default, so we back the target-mounts by tmpfs.
+            self.mount(logger).await?;
+
+            // Spawn background thread to monitor changes
+            let join_handle = Self::spawn_watcher(
```

Comment:
nit: 
```Rust
self.watch_thread = Some(Self::spawn_watcher(
...
```

Reply:
The cost of variables and lines in rust land must be hella expensive :D

Reply:
The Rust toolchain makes you so dumb that you want to pretend you still know better than LLVM and want to help it. I'm sure I'm saving one register renaming here, it's crucial.